### ISSUE TITLE
Running multiple instances of Codemirror/Ace runs into problems

### ISSUE THREAD

**Issue by hkirat** [y-js/y-text#8]
Hi
I was trying to run multiple instances of editors for Codemirror/Ace.
The program doesn't seem to work as intended.

Here is what I observed.
1. Using Ace
Including multiple Ace editors in the same window makes them relay events amongst themselves.
As a result, a change in any window is reflected in the other. They are all initialized to the same value as well. 
I have defined different editors and called the bind function separately on the two.
![vid](https://media.giphy.com/media/l4FGsrjPBNJnVhNug/giphy.gif)

2. Using Codemirror
Something worse happens in this case. The control seems to be stuck among the functions 
```js
codeMirrorCallback
```
and 
```js
yCallback
```
They keep on calling each other and eventually the window crashes.

I am not sure if the code of y-text was written to handle multiple instances of Codemirror/Ace.
It seems though it should since the code consists of [arrays](https://github.com/y-js/y-text/blob/master/src/Text.js#L97)
```js
codeMirrorInstances
```
and
```js
aceMirrorInstances
```
which suggests multiple instances should be handled

If not then this issue can be converted into a discussion on how to do that.

**Comment by dmonad** [comment]
Hi @hkirat,

It should be possible to bind multiple editor instances to the same / different Yjs instances. Since events in the Ace editor fire synchronously, I expected CodeMirror to work the same way. I think I found the right event now. So your second issue is fixed in y-text@9.3.2.

Your first point describes the expected behavior. If you want several editor instances with different content, you have to define more types. E.g.

```javascript
Y({
  ..
  share: {
    editor1: 'Text', // y.share.editor1 is of type Y.Text
    editor2: 'Text' // y.share.editor2 is of type Y.Text
  }
}).then(function (y) {
  var cm1 = CodeMirror(document.body)
  var cm2 = CodeMirror(document.body)
  y.share.editor1.bindCodeMirror(cm1)
  y.share.editor2.bindCodeMirror(cm2)
})
```

Here, `y.share.editor1` is the source of truth for the editor content. The previous content (before binding) of the CodeMirror instance is always overwritten.

**Comment by hkirat** [comment]
Works like a charm.
Thanks for the fix @dmonad
